first draft of onboarding doc (#186)
BCDevOps/developer-experience: docs/OnboardingOutcomes.md

---
resourceType: Documentation
personas: 
  - Developer
  - Product Owner
  - Designer
tags:
  - openshift
  - new
  - onboarding
  - project set
  - devops
description: Learn where to ask for help with issues not covered by devops-requests
---

# Onboarding Outcomes

This document is primarily for the use of those teams who have already had an onboarding meeting with the Platform Services team,
but it contains plenty of useful information for everyone else, too!

## Our Expectations for You

As one of the vanguard of openness and agility in government, your presence on the Openshift Platform comes with some expectations.
At their core, these expectations boil down to one thing: **Leadership**. You're here to explore a new, more digital government.
You're building the path that others will follow. That means that you will be expected to help others to follow that path.

### A Community Supported Platform

> In short, this just means one thing: help each other. If you're having problems - with *anything* to do with being a DevOps team - your first stop for help should be the community.

Reviewer: ```suggestion
In short, this just means one thing: help each other. If you're having problems - with *anything* to do with being a DevOps team - **your first stop for help should be the community.**
```

This refers to both the Developer Exchange community, and the larger development community on the web.

> When you're having a problem, your first stop should be to seek out the rest of the community. This probably means RocketChat (#devops-how-to is always a good place to start), and can also mean seeking out help from the community that exists around your chosen open-source tools on the wider web, as well.

Reviewer: ```suggestion
When you're having a non-urgent question or a problem, your first stop should be to seek out the rest of the community.  RocketChat's #devops-how-to channel is always a good place to start.  You can also seek out help from the community that exists around your chosen open-source tools on the wider web, as well.
```

> It also means that you should be keeping a close eye on rocketchat yourself, and offering support to other teams as they encounter problems. It's a two-way street!

Reviewer: ```suggestion
It also means that you should be keeping a close eye on Rocketchat yourself, and offering support to other teams as they encounter problems. It's a two-way street!
```

If you're from elsewhere in government, this is probably a process that is different from what you're used to - normally, you'd call 7-7000 and someone else would swing in to fix it for you.
Self-service cuts both ways: you have all kinds of freedom and flexibility, but it also means responsibility. The more you provide your support to others, the more others will be able to support you.

However, if you suspect a problem with *the platform itself* (such as an outage), head on over to [our status page](https://status.pathfinder.gov.bc.ca) or #devops-alerts and/or #general on RocketChat to see if we are already aware of a platform-wide problem. 
If you don't see anything there, then #devops-operations is a good place to start. However, you should always be *sure* that it's a platform issue before posting there. If you're not sure, start on #devops-how-to to see if you can get some help with that.

### An Open Platform

Open is the name of the game. If it can be open, it should be. If it can't be open, you should aim to fix it so that it can.

> All of our teams' code is open-source and available on the bcgov github organization.

Reviewer: ```suggestion
All of our teams' code is open-source and available on the bcgov github organization with the exception of a few highly sensitive projects that for the time being keep their code in private repos until they figure out how to make it open to public in a safe manner.
```

Author: To me, this suggests exceptions rather than just apps that are on the path to becoming open. What about this:
```suggestion
All of our teams' code is either in open-source and available on the bcgov github organization, or is (in rare cases) in the process of adjusting their sensitive code in a manner that would allow them to move their repository to the open-source bcgov organization within the short or medium term.
```

You will be required to have a RocketChat account before you are able to do much of anything else on the platform, so don't delay.

### Your Team

When you join the platform and the exchange, you should have (or be working on getting) a team that is able to follow the agile process that are core to our work here.
Every team on the platform *must* have a Product Owner and a Technical Steward (and these people should both be BCGov employees whenever possible).
Your team also *should* have a Scrum Master and at least one DevOps Specialist on board. These aren't strictly *required*, but long experience does tell us that you're going to have a Bad Time if you don't have these positions filled.

## Your Expectations for Us

We are here to help you as much as we can, but we take the community-supported self-service model very seriously, and so here we will discuss what you should (and shouldn't!) expect the Platform Services Team to provide for you.

### Community-Wide Tools

One of our primary mandates is to provide the whole community with a suite of tools that will allow you to make full use of the platform. At the moment, these include:

> * RocketChat

Reviewer: ```suggestion
* RocketChat (https://chat.pathfinder.gov.bc.ca or download an app)
```

> * Keycloak

Reviewer: ```suggestion
* Keycloak Single Sign-On
```

> * Aporeto

Reviewer: ```suggestion
* Aporeto (coming in the new OCP 4 Platform in summer 2020)
```

> * Documize

Reviewer: ```suggestion
* Documize Document Management (https://docs.pathfinder.gov.bc.ca/) - supports public and protected content
```

> * Github organizations (bcgov and BCDevOps)

Reviewer: ```suggestion
* GitHub - online code management for open-source projects, all government repos are part of  the **bcgov** organization.
```

> * Keycloak Single Sign-On and Federated Identity Service (https://sso.pathfinder.gov.bc.ca/), on-prem hosted

Reviewer: ```suggestion
* Keycloak Single Sign-On and Federated Identity Service (https://sso.pathfinder.gov.bc.ca/ and https://developer.gov.bc.ca/BC-Government-SSO-Service-Definition#features--functions), on-prem hosted
```

* ... and obviously the platform itself, Openshift.

We are also working on rolling out new tools for the community to use. We are always interested in hearing recommendations for more - just make sure that, whatever you suggest, it's something that would be useful to the *community* and not just your team!

### Platform Support

We support the platform, making sure that it works properly and that it is available for everyone to use. It is *not* tech support for the platform. Here's a good example of the difference:

> * If you're having problems getting your Rocketchat application to work, that's a *community* support issue.

Reviewer: ```suggestion
* If you're having problems getting your RocketChat application to work, that's a *community* support issue. Reach out to the community for help.
```

> * If you suspect that the RocketChat service is having an outage, that's a *platform* support issue.

Reviewer: ```suggestion
* If you suspect that the RocketChat service is having an outage, that's a *platform* support issue. Check out the Platform Services Status Page (https://status.pathfinder.gov.bc.ca) for outage updates.
```

It is our responsibility to make sure that the services we provide are available and stable.

### Common Components

Part of the community support model also means that teams can and should make sure that they're not re-inventing the wheel.
If your team makes a login script, it's very likely that such code would be useful to other teams. However, we also understand that making that code modular and generalized for use by other teams isn't always easy.
To that end, we have a whole team dedicated to the creation of such components, as well as supporting teams who want to provide easy-to-use components to other teams.

> You can find them at #team-coco on RocketChat

Reviewer: ```suggestion
You can find them at #team-coco on RocketChat. They will be happy to work with you to help you find an existing component that meets your needs or help you refactor the code specific to your app into a common component that other teams can benefit from.
```

---
resourceType: Documentation
personas: 
  - Developer
  - Product Owner
  - Designer
tags:
  - openshift
  - new
  - onboarding
  - project set
  - devops
description: Learn where to ask for help with issues not covered by devops-requests
---

# Welcome to our Platform Community! 

Glad you’re here! If you’ve already had your onboarding meeting with the Platform Services team, you’ll find this follow-up info helpful as part of your welcome package to our community. If you’re just scoping us out, we hope you like what you learn and will come join our journey.  

## We’re Better Together

Dreamers. Vanguards. Innovators. We’re all creating lasting change by moving our government into a brave new world of openness and agility. **We can’t do it alone and we don’t pretend we can.** So, we’re holding hands and setting off on an adventure together to go towards something greater.

You’re here to explore a new, more digital government. You’re building the path that others will follow. You’re the leader that you always wish you had, and with the right people by your side, you’ll have the courage to keep going.

Try. Experiment. Iterate. Fail. Try again. Change the world, together.

> ## What this means for YOU

Reviewer 2: What this means for you

In short, it means that the value you bring to our community is not what you do for yourself. **Your value is determined by what you do for others**. You get to work with a sense of purpose, cause and belief as you contribute to something even bigger than your own project. Not bad for a paid gig.

### A Community-Supported Platform

Our Platform is supported by our community. Quite simply, this means: we help each other. **Your first stop for help should always be the community**. This refers to both the Developer Exchange community, and the larger development community on the web.

* Don’t know what to do? Say it out loud to our community and chances are that someone who does know will offer help.

> * Holding onto a great idea, or have a brilliant answer? Give it away, and watch it become even better.  

Reviewer 2: * Holding onto a great idea, or have a brilliant answer? Contribute it.  Share it.  You'll be amazed at how quickly the community will join you in your journey

You’ll need a RocketChat account before you’re  able to do much of anything else on the Platform, so don't delay.

### Your Team

When you join the Platform and the Exchange, you should have (or be working on getting) a team that is able to follow the agile process that are core to our work here. 

Every team on the Platform must have a **Product Owner** and a **Technical Steward** (and these people should both be BCGov employees whenever possible). 

Your team also should have a **Scrum Master** and at least one **DevOps Specialist** on board. (These aren't strictly required, but long experience does tell us that you're going to have a Bad Time if you don't have these positions filled.)

## What’s coming your way from US 

We’re here for you as part of the Platform Services Community. We foster the love in our community-supported self-service model—and take it seriously—so let’s chat about what you should (and shouldn't!) expect the Platform Services Team to provide.

### Community-Wide Tools

One of our primary mandates is to provide the whole community with a suite of tools that will allow you to make full use of the Platform. At the moment, these include:

> * RocketChat (https://chat.pathfinder.gov.bc.ca or download an app), on-prem hosted

Reviewer: ```suggestion
* <a target="_blank" href="https://github.com/bcgov/devhub-resources/blob/master/resources/community/rocketchat.md">RocketChat </a> (https://chat.pathfinder.gov.bc.ca or download an app), on-prem hosted
```

> * Keycloak Single Sign-On and Federated Identity Service (https://sso.pathfinder.gov.bc.ca/), on-prem hosted

Reviewer: ```suggestion
* <a href="https://developer.gov.bc.ca/BC-Government-SSO-Service-Definition" target="_blank"> Keycloak Single Sign-On and Federated Identity Service <a/> (https://sso.pathfinder.gov.bc.ca/), on-prem hosted
```

> * Aporeto (coming in the new OCP 4 Platform in summer 2020), on-prem and SaaS components

Reviewer: ```suggestion
* <a href="https://developer.gov.bc.ca/Developer-Guide-to-Zero-Trust-Security-Model-on-the-Platform" target="_blank">Aporeto </a> (coming in the new OCP 4 Platform in summer 2020), on-prem and SaaS components
```
